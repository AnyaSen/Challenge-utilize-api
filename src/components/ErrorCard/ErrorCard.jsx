import React from 'react';
import Styles from './ErrorCard.module.scss';

import errorIcon from '../../assets/img/error.svg';

export default function ErrorCard() {
	return (
		<div data-testid='ErrorCard' className={Styles.ErrorCardContainer}>
			<img src={errorIcon} alt='Error' />

			<p data-testid='errorMessage'>Sorry, something went wrong. :(</p>
		</div>
	);
}
