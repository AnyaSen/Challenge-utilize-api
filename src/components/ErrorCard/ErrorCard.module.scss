@import '../../assets/scss/variables';
@import '../../assets/scss/breakpoints';

.ErrorCardContainer {
	display: flex;
	flex-direction: column;
	border: 1px solid $color-red;
	border-radius: $border-radius;
	padding: $space-5 $space-7;
	max-width: 45rem;
	margin: $space-2;

	img {
		height: 9rem;
		margin-bottom: $space-4;
	}

	p {
		text-align: center;
	}

	@include md {
		padding: $space-5;
	}

	@include sm {
		padding: $space-4;

		img {
			height: 6rem;
		}
	}
}
