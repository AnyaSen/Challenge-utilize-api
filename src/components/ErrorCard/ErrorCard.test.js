import React from 'react';
import { render, cleanup } from '@testing-library/react';

import ErrorCard from './ErrorCard';

describe('<ErrorCard />', () => {
	afterAll(cleanup);

	test('Should contain the correct image & text', () => {
		const { getByTestId, getByAltText, container } = render(<ErrorCard />);

		const errorMessage = 'Sorry, something went wrong. :(';

		const errorIcon = getByAltText('Error');
		const errortext = getByTestId('errorMessage');

		expect(errorIcon.src).toContain('error.svg');
		expect(errortext.textContent).toBe(errorMessage);

		expect(container.firstChild).toMatchSnapshot();
	});
});
