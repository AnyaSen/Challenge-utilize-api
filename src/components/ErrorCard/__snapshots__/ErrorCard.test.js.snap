// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorCard /> Should contain the correct image & text 1`] = `
<div
  class="ErrorCardContainer"
  data-testid="ErrorCard"
>
  <img
    alt="Error"
    src="error.svg"
  />
  <p
    data-testid="errorMessage"
  >
    Sorry, something went wrong. :(
  </p>
</div>
`;
